### DISCUSSION TITLE
Delete/remove file from batch

### DISCUSSION THREAD

**Discussion by asynxc** [rpldy/react-uploady#212] [Q&A]
**Is your feature request related to a problem? Please describe.**
a possible way to remove file from a batch.
after adding files (especially a large amount of files) users may want to remove some of them before event starting the upload process. and unfortunately, there's no way to do it. i think it will be possible if the internal state was exposed.

**Describe the solution you'd like**
add `delete` function to the api that allow remove files from batchs if those files are not in `FILE_STATES.UPLOADING` state

**Comment by yoavniran** [comment]
Hey @hamavb 

There are at least a couple of ways to remove files from Uploady's batch.

There is the simple [File Filter](https://github.com/rpldy/react-uploady/blob/master/guides/FileFilter.md) method, this is the way to remove automatically based on application logic. 

Then there is the way to do it using what I call the Queue approach. The term I used is "abort" but for the end-user this can be made to behave and look the same as delete/remove. 
You can refer to these resources for details and code examples:

[UploadQueue Guide](https://github.com/rpldy/react-uploady/blob/master/guides/UploadQueue.md)
[hooked on file uploads with react-uploady](https://dev.to/poeticgeek/hooked-on-file-uploads-with-react-uploady-3h0j)
[react-uploady-with-ant-design](https://dev.to/poeticgeek/react-uploady-with-ant-design-dja)

As I mentioned, the "delete function" is actually implemented as "abort" using the [useAbortItem](https://github.com/rpldy/react-uploady/tree/master/packages/ui/uploady#useabortitem), [useAbortBatch](https://github.com/rpldy/react-uploady/tree/master/packages/ui/uploady#useabortbatch), and [useAbortAll](https://github.com/rpldy/react-uploady/tree/master/packages/ui/uploady#useabortall) **hooks**.

**Comment by yoavniran** [comment]
moving to discussions
